src/effects/builtin/reverbeffect.cpp

```diff
@@ -73,14 +76,31 @@ EffectManifestPointer ReverbEffect::getManifest() {
     send->setMinimum(0);
     send->setDefault(0);
     send->setMaximum(1);
+
+    EffectManifestParameterPointer dryWet = pManifest->addParameter();
+    dryWet->setId("dry_wet");
```

Comment:
What is the use-case of this parameter? Can't we follow the echo implementation?

Reply:
https://github.com/mixxxdj/mixxx/pull/1676#issuecomment-390804177

Reply:
> It turns out that a dry/wet parameter is needed for the Reverb effect. Before now the effect Was always fully wet with no dry, with the amount of the effect control with the Send parameter. 

No,It was always "out[i] = xl + in[i];"
The dry_wet knob makes it worse, now and is redundant with the Mix knob.

Reply:
> To test this I recommend playing with the chain BQ EQ -> Echo -> Reverb.

You can argue the same for the revers order, but the Echo has no Dry/Wet Knob. 
I think this is a case that musically does not make much sense, compared to the GUI clutter. 
We should stick with one concept. If it is really a desired effect, my preferred solution is to put it in a dedicated new effect.

Reply:
All return effects like echo and reverb, do there effect by taking the dry signal, alter it and put it back, on top of the dry channel. If we chain a echo, and put the  reverb into the delay path, it has no access to the dry signal and cannot work as return effect any more. Musically it might may sense to add it as insert effect here. In this case the reverb has to pass through the delayed signal as 100% and adds the reverb onto it. This can be controlled by the send button, and might be additional controlled by a mix knob at the output.

The same is true if you put the reverb first. I don't know if it makes sense to put an echo after but if the echo eats all non delayed signal, the reverb effect is gone.

So it looks like that the effect mode must only be consumed from one effect in the chain to allow the most desired effect without overcomplicate the GUI.

Reply:
> No,It was always "out[i] = xl + in[i];"

Whoops, you are right. I think I got confused because the Echo effect's Send parameter acts as a redundant volume knob for the whole chain in D+W mode, but with Reverb mixed as `out[i] = xl + in[i]` that is not the same.

Reply:
So I think we can simply remove the new Dry/Wet parameter for Reverb.

Reply:
Yes. 
We need to decide if we need something else.
Should reverb follow the mix mode like echo? Is there a use case to add a filter to the reverb wet path?
If yes, I think only the first effect in the chain should switch to return mode.

Reply:
Looking back at the [diff of the commit where I introduced the new Dry/Wet parameter](https://github.com/mixxxdj/mixxx/pull/1676/commits/fd25553f2905c07165980084914ed648fe6dfa81#diff-ac5b3c7153b967ecc275913aafed81d6L457), I got confused because I had changed `out[i] = xl + in[i];` to `xr + (addDry ? in[i + 1] : 0);` in 2b72491ee2a0bd761e0d68600b62ebbb3b89dc6c. However, I didn't need to do anything to Reverb in 2b72491ee2a0bd761e0d68600b62ebbb3b89dc6c. Simply keeping `out[i] = xl + in[i];` regardless of the effect unit mix mode as it was before works.